### ISSUE TITLE
Is it possible to get the "angled" margin of a glyph in an italic font?

### ISSUE THREAD

**Issue by arrowtype** [fonttools/ufoLib2#280]
RoboFont has the very useful glyph attributes `glyph.angledLeftMargin` and `glyph.angledRightMargin` in its API. This is pretty vital when scripting Italic fonts. If I just ask for the basic left/right margin, it is “wrong” in terms of how I need to manipulate a glyph drawing.

Is there an existing function  to get `glyph.angledLeftMargin` and `glyph.angledRightMargin`, or object with these attributes (in ufoLib2 or elsewhere, but independent of a font editor)?

As a specific example, if I try to position an `ogonek` anchor on an italic /E, I am unsure of how to determine the bottom right corner... even though it is the further-right point, if the margin is angled. Even if I use the `font.lib["com.typemytype.robofont.italicSlantOffset"]` of the UFO, it is still way off, because using the glyph bounds or basic rightMargin value is based on the top right corner. (This makes sense, because the font is _ultimately_ just in a rectangle, and the angled margins are just there to help designers draw italic fonts in a way that makes sense... but when I’m coding, I frequently wish I had a way to do this, too!)

<img width="963" alt="image" src="https://github.com/fonttools/ufoLib2/assets/45946693/44ebe6df-9519-435d-b167-e48cc9924356">

Thanks for any tips or pointers, if you happen to have them!

**Comment by anthrotype** [comment]
> Is there an existing function

no, but we could add one I suppose, if needed. Does defcon or fontParts have something like that that we can borrow?

**Comment by arrowtype** [comment]
Unfortunately, no, neither defcon nor fontParts have something like that.
